src/main/java/org/broadinstitute/hellbender/tools/copynumber/arguments/CopyNumberArgumentValidationUtils.java

```diff
@@ -184,4 +187,82 @@ public static AnnotatedIntervalCollection validateAnnotatedIntervalsSubset(final
                 "Annotated intervals do not contain all specified intervals.");
         return new AnnotatedIntervalCollection(locatableCollection.getMetadata(), subsetAnnotatedIntervals);
     }
+
+    /**
+     * Validate that input files and/or directories are readable if they are not {@code null} (i.e., optional inputs).
+     */
+    public static void validateInputs(final File ... inputs) {
+        if (inputs != null) {
+            for (final File input : inputs) {
+                if (input != null) {
+                    if (input.isFile()) {
+                        IOUtils.canReadFile(input);
+                    } else if (input.isDirectory() && !input.canRead()) {
+                        throw new UserException.CouldNotReadInputFile(input);
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Validate that output files are writeable, whether or not they already exist.
+     */
+    public static void validateOutputFiles(final File ... outputs) {
+        Utils.nonNull(outputs);
+        for (final File output : outputs) {
+            Utils.nonNull(output);
+            if ((output.exists() && !output.canWrite()) || (!output.exists() && !output.getAbsoluteFile().getParentFile().canWrite())) {
```

Comment:
Assuming this works on directories, if `output` is root (`/`) then `getParentFile()` returns null and this'll throw a null pointer exception.

Reply:
See below.